docs/v3/core/v3.0.rst

```diff
@@ -895,6 +875,65 @@ mandatory names:
     must be a string referring to a v3 chunk grid specification. The
     ``configuration`` is optional and defined by the extension.
 
+``chunk_key_encoding``
+^^^^^^^^^^^^^^^^^^^^^^
+
+    The mapping from chunk grid cell coordinates to keys in the underlying
+    store.
+
+    The value must be an object with required string member ``name``, specifying
+    the encoding type, and optional member ``configuration`` specifying encoding
+    type-dependent parameters; the ``configuration`` value must be an object if
+    it is specified.
+
+    The following encodings are defined:
+
+    - ``default``
+
+      The ``configuration`` object may contain one optional member,
+      ``separator``, which must be either ``"/"`` or ``"."``.  If not specified,
+      ``separator`` defaults to ``"/"``.
+
+      The key for a chunk with grid index (``k``, ``j``, ``i``, ...) is
+      formed by taking the initial prefix ``c``, and appending for each dimension:
+
+      - the ``separator`` character, followed by,
+
+      - the ASCII decimal string representation of the chunk index within that dimension.
+
+      For example, in a 3 dimensional array, with a separator of ``/`` the identifier
+      for the chunk at grid index (1, 23, 45) is the string ``"c/1/23/45"``.  With a
+      separator of ``.``, the identifier is the string ``"c.1.23.45"``.
+
+      .. note:: A main difference with spec v2 is that the default chunk separator
+         changed from ``.`` to ``/``, as in N5.  This decreases the maximum number of
+         items in hierarchical stores like directory stores.
+
+      .. note:: Arrays may have 0 dimensions (when for example representing scalars),
+         in which case the coordinate of a chunk is the empty tuple, and the chunk key
+         will consist of the string ``c``.
+
+    - ``v2``
+
+      The ``configuration`` object may contain one optional member,
```

Comment:
Maybe add a short hint that this is purely meant for backwards-compatibility (e.g. only writing new metadata).

Reply:
Done.